android/iden3coreapi/src/main/java/com/iden3/iden3coreapi/Iden3Identity.kt

```diff
@@ -0,0 +1,155 @@
+package com.iden3.iden3coreapi
+
+import iden3mobile.*
+import java.util.ArrayList
+import java.util.HashMap
+
+class Iden3Identity(private var identity: Identity) {
+
+    /**
+     * Stop an Iden3 identity.
+     *
+     */
+    fun stopIdentity() {
+        identity.stop()
+    }
+
+    /**
+     * Request a claim from an issuer.
+     *
+     * @param issuerUrl: Url of the issuer.
+     * @param data: data needed by the issuer for requesting a claim.
+     * @param ticketListener: Listener of tickets received after requesting the claim.
+     *
+     * @return An Iden3Ticket.
+     */
+    @Throws(Exception::class)
+    fun requestClaim(issuerUrl: String, data: String, ticketListener: Iden3TicketListener?) : Iden3Ticket? {
+        if (ticketListener == null) {
+            return Iden3Ticket(identity.requestClaim(issuerUrl, data))
+        } else {
+            identity.requestClaimWithCb(issuerUrl, data) {
+                ticket, exception ->
+                if (exception != null) {
+                    ticketListener.onError(exception)
+                } else if (ticket != null) {
+                    ticketListener.onTicketReceived(Iden3Ticket(ticket))
+                }
+            }
+        }
+        return null
+    }
+
+    /**
+     * Prove a claim from an verifier.
+     *
+     * @param verifierUrl: url of the verifier.
+     * @param credentialId: credential id needed by the verifier for proving a claim.
+     * @param withZKProof: Boolean that indicates if zero-knowledge proof shoud be used or not.
+     * @param proveClaimListener: Listener of proofs received after proving a claim.
```

Comment:
The proofs are not received but generated on device. What is received is a boolean response, true if the verifer has been able to verify the proof, false otherwise.

Reply:
done